### ISSUE TITLE
Bug: Local clustering coefficient on a multi-graph

### ISSUE THREAD

**Issue by tomasonjo** [neo4j/graph-data-science#65]
I have noticed that the local clustering coefficient behaves weirdly on a multigraph projection. Perhaps it should always be reduced to single graph before execution?

Using Neo4j 4.0.5 and GDS 1.2.2, but I noticed this error when playing around with 1.3.0alpha2 GDS

Create sample graph:

```
CREATE (a:Node)-[:REL]->(b:Node)-[:REL]->(c:Node)-[:REL]->(a)
WITH a,b
UNWIND range(1,7) as r
CREATE (a)-[:REL]->(b)
```

Load the graph in memory:

`CALL gds.graph.create('test','*',{type:{type:'*', orientation:'UNDIRECTED'}})`

Execute algorithm:

```
CALL gds.localClusteringCoefficient.stream('test')
YIELD nodeId, localClusteringCoefficient
RETURN nodeId, localClusteringCoefficient ORDER BY 
localClusteringCoefficient DESC LIMIT 10
```
returns

nodeId | localClusteringCoefficient
-- | --
2 | 4.0
1 | 0.1111111111111111
0 | 0.1111111111111111

Given that the local clustering coefficient should be between 0 and 1, this can be very confusing to end user.

**Comment by Mats-SX** [comment]
This one is a bit tricky. The calculation for LCC is 

(number of triangles * 2) / degree * (degree - 1)

When parallel edges exist, the degree increases but the number of triangles does not. This is because in our definition, a triangle is a triple of three nodes, so any one relationship between the nodes is enough and extra ones make no difference.

If we do not want to challenge this definition of a triangle, we unfortunately do not have any ability to find the _unique degree_ of a node, ie the degree of the node where each unique neighbour is counted once, rather than a neighbour existing multiple times being counted multiple times. So in this case we could restrict LCC to run only on graphs without any parallel relationships.

**Comment by tomasonjo** [comment]
Do we actually know if we loaded any parallel relationships in the graph? 
IMO it's better to restrict a graph algorithm usage than to let the user run it not knowing that it might provide inaccurate results. Especially as this is production-ready tier.

If we ever decide to upgrade the link prediction algos to use the in-memory graph, we could then also fix this issue as for example, the Common Neighbours algo will need to know which node is on the other side of the relationship.

**Comment by Mats-SX** [comment]
@tomasonjo Thanks for reporting this. We solved this issue in https://github.com/neo4j/graph-data-science/commit/c684b18ed0e67adda4125cbfd2a36b00d339b4df. It is included in the `1.2.3` release and will make it into `1.3.1` when that comes out.
